src/extension-runners/chromium.js

```diff
@@ -292,10 +298,30 @@ export class ChromiumExtensionRunner {
   async reloadExtensionBySourceDir(
     extensionSourceDir: string // eslint-disable-line no-unused-vars
   ): Promise<Array<ExtensionRunnerReloadResult>> {
-    // TODO(rpl): detect the extension ids assigned to the
-    // target extensions and map it to the extensions source dir
-    // (https://github.com/mozilla/web-ext/issues/1687).
-    return this.reloadAllExtensions();
+    const runnerName = this.getName();
+    const extensionId = this.generateId(extensionSourceDir);
+
+    await this.wssBroadcast({
+      type: 'webExtReloadExtension',
+      extensionId,
+    });
+
+    return [{runnerName}];
+  }
+
+  /**
+   * Generate extensionId from extension sourceDir
+  **/
+  generateId(sourceDir: string): string {
+    const hexHash = crypto.createHmac('sha256', sourceDir).digest('hex');
```

Comment:
You need to normalize the path. The `sourceDir` value needs to be in the same format as Chrome's internal `FilePath`, and on Windows the paths need to be normalized as well: https://cs.chromium.org/chromium/src/components/crx_file/id_util.cc?l=61&rcl=9e60666b0869fb07b4b6f92947d54f31fda1fb70&targetos=win

This directory-based ID-generation is difficult to get right. Are you sure that you want to go down this road?